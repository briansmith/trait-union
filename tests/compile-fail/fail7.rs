use trait_union::trait_union;

trait F { }

impl F for &str { }

trait_union! {
    union U<'a>: F+'a = &'a str;
}

fn f(s: &'_ str, u: &mut U<'_>) {
    *u = U::new(s);
}

fn main() {
}
