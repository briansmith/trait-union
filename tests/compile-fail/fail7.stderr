error[E0623]: lifetime mismatch
  --> $DIR/fail7.rs:12:10
   |
11 | fn f(s: &'_ str, u: &mut U<'_>) {
   |         -------          -----
   |         |
   |         these two types are declared with different lifetimes...
12 |     *u = U::new(s);
   |          ^^^^^^^^^ ...but data from `s` flows into `u` here
